runtime/src/main/java/io/quarkiverse/fury/FuryProducer.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkiverse.fury;
+
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Singleton;
+
+import org.apache.fury.Fury;
+
+/**
+ * Producers of beans that are injectable via CDI.
+ */
+@Singleton
+public class FuryProducer {
+    private volatile Fury fury;
```

Comment:
Fury is not thread safe, we may need to use ThreadSafeFury

Reply:
OK, it can be configurable. We can add such property like
```
quarkus.fury.thread-safe=true
```

Reply:
Fury is a singleton here, do we have cases to use a single-threaded Fury in quarkus?

Reply:
TBH, I'm not sure what is the difference with the single-threaded Fury. Feel free to open an issue. We can not resolve all of things in on PR :)

Reply:
You can't use `Fury` object to serialize objects at multiple threads at the same time. `Fury` doesn't work that way, it can be used to serialize an object once a time. To use `Fury` object serialize multiple objects at different threads, users must use `ThreadSafeFury`. I think for singleton object here, it can only be `ThreadSafeFury`.

Reply:
Yeah, I understand. Please add such a note in the `quarkus-fury` documentation. And also with `quarkus.fury.thread-safe` property. If it set to `false`, `Fury` can not be used in the multithreads.